rules/prefer-global-this.js

```diff
@@ -0,0 +1,494 @@
+'use strict';
+
+const MESSAGE_ID_ERROR = 'prefer-global-this/error';
+const MESSAGE_ID_SUGGESTION = 'prefer-global-this/suggestion';
+const messages = {
+	[MESSAGE_ID_ERROR]: 'Prefer `{{replacement}}` over `{{value}}`.',
+	[MESSAGE_ID_SUGGESTION]: 'Replace `{{value}}` with `{{replacement}}`.',
+};
+
+/**
+ * Find the variable in the scope
+ * @param {import('eslint').Scope.Scope} scope
+ * @param {string} variableName
+ * @returns
+ */
+function findVariableInScope(scope, variableName) {
+	if (!scope || scope.type === 'global') {
+		return;
+	}
+
+	const variable = scope.variables.find(v => v.name === variableName);
+
+	if (variable) {
+		return variable;
+	}
+
+	return findVariableInScope(scope.upper, variableName);
+}
+
+const globalIdentifier = new Set(['window', 'self', 'global']);
+
+const windowSpecificAPIs = new Set([
+	// Refs: https://html.spec.whatwg.org/multipage/nav-history-apis.html#the-window-object
+	// Properties and methods
+	'window',
+	'self',
+	'document',
+	'name',
+	'location',
+	'history',
+	'navigation',
+	'customElements',
+	'locationbar',
+	'menubar',
+	'personalbar',
+	'scrollbars',
+	'statusbar',
+	'toolbar',
+	'status',
+	'close',
+	'closed',
+	'stop',
+	'focus',
+	'blur',
+	'frames',
+	'length',
+	'top',
+	'opener',
+	'parent',
+	'frameElement',
+	'open',
+	'object',
+	'navigator',
+	'clientInformation',
+	'originAgentCluster',
+	'alert',
+	'confirm',
+	'prompt',
+	'print',
+	'postMessage',
+
+	// global events
```

Comment:
Not sure these should be here? They don't seem window-specific.

Reply:
It is difficult to determine which properties or methods are window-specific.

But Federico had updated it. I think is have been resolved.